src/main/java/com/google/devtools/build/lib/remote/common/RemoteCacheClient.java

```diff
@@ -63,6 +65,26 @@ public int hashCode() {
     }
   }
 
+  @AutoValue
+  abstract class CachedActionResult {
+    public static CachedActionResult create(ActionResult actionResult, String cacheName) {
+      return new AutoValue_RemoteCacheClient_CachedActionResult(actionResult, cacheName);
+    }
+
+    public static CachedActionResult remote(ActionResult actionResult) {
+      return new AutoValue_RemoteCacheClient_CachedActionResult(actionResult, "remote");
+    }
+
+    public static CachedActionResult disk(ActionResult actionResult) {
+      return new AutoValue_RemoteCacheClient_CachedActionResult(actionResult, "disk");
+    }
+
+    @Nullable
+    public abstract ActionResult actionResult();
```

Comment:
Please add java doc to describe in which case the value will be `null`.

Reply:
Changed to be non-nullable